app/src/main/java/com/github/gotify/service/WebSocketService.kt

```diff
@@ -0,0 +1,376 @@
+import com.github.gotify.messages.Extras
+import com.github.gotify.messages.MessagesActivity
+import com.github.gotify.picasso.PicassoHandler
+import com.github.gotify.service.WebSocketConnection.BadRequestRunnable
+import com.github.gotify.service.WebSocketConnection.OnNetworkFailureRunnable
+import io.noties.markwon.Markwon
+import java.util.concurrent.atomic.AtomicLong
+
+internal class WebSocketService : Service() {
+    companion object {
+        val NEW_MESSAGE_BROADCAST = "${WebSocketService::class.java.name}.NEW_MESSAGE"
+        private const val NOT_LOADED = -2L
+    }
+
+    private lateinit var settings: Settings
+    private var connection: WebSocketConnection? = null
+
+    private val lastReceivedMessage = AtomicLong(NOT_LOADED)
+    private lateinit var missingMessageUtil: MissedMessageUtil
+
+    private lateinit var picassoHandler: PicassoHandler
+    private lateinit var markwon: Markwon
+
+    override fun onCreate() {
+        super.onCreate()
+        settings = Settings(this)
+        val client = ClientFactory.clientToken(
+            settings.url,
+            settings.sslSettings(),
+            settings.token
+        )
+        missingMessageUtil = MissedMessageUtil(client.createService(MessageApi::class.java))
+        Log.i("Create ${javaClass.simpleName}")
+        picassoHandler = PicassoHandler(this, settings)
+        markwon = MarkwonFactory.createForNotification(this, picassoHandler.get())
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        if (connection != null) {
+            connection!!.close()
+        }
+        Log.w("Destroy ${javaClass.simpleName}")
+    }
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        Log.init(this)
+        if (connection != null) {
+            connection!!.close()
+        }
+        Log.i("Starting ${javaClass.simpleName}")
+        super.onStartCommand(intent, flags, startId)
+        Thread { startPushService() }.start()
+
+        return START_STICKY
+    }
+
+    private fun startPushService() {
+        UncaughtExceptionHandler.registerCurrentThread()
+        showForegroundNotification(getString(R.string.websocket_init))
+
+        if (lastReceivedMessage.get() == NOT_LOADED) {
+            missingMessageUtil.lastReceivedMessage { lastReceivedMessage.set(it ?: 0L) }
+        }
+
+        val cm = getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager
+        val alarmManager = getSystemService(ALARM_SERVICE) as AlarmManager
+
+        connection = WebSocketConnection(
+            settings.url,
+            settings.sslSettings(),
+            settings.token,
+            cm,
+            alarmManager
+        )
+            .onOpen { onOpen() }
+            .onClose { onClose() }
+            .onBadRequest { message -> onBadRequest(message) }
+            .onNetworkFailure { minutes -> onNetworkFailure(minutes) }
+            .onMessage { if (it != null) onMessage(it) }
```

Comment:
```suggestion
            .onMessage { onMessage(it) }
```

Reply:
Resolved in aeddf508751cdf908d177b070ef334918c643bf2